src/main/java/uk/gov/companieshouse/api/accounts/controller/CurrentPeriodController.java

```diff
@@ -100,6 +105,55 @@ public ResponseEntity create(@RequestBody @Valid CurrentPeriod currentPeriod,
         return responseEntity;
     }
 
+    @PutMapping
+    public ResponseEntity update(@RequestBody @Valid CurrentPeriod currentPeriod,
+        BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+        HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.CURRENT_PERIOD.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult, errors);
+        }
+
+        currentPeriodValidator.validateCurrentPeriod(currentPeriod, errors);
```

Comment:
isn't this the responsibility of the service rather than the controller?

Reply:
Again this will form part of a larger look at validation.